### PR TITLE
short out of rekey flow if our device is in the list of solutions

### PR DESCRIPTION
**Pull request by maxtaco**
- do so both in the initial entry and also in the retry loop
- also fix a bug in deciding if our device solves the problem. it's not
  good enough if our device is in the device list, we have to check that it
  solves all of the given problems.

### CODE REVIEW

File: go/service/rekey_ui_handler.go [hunk 1]
```diff
@@ -103,9 +103,54 @@ func (r *RekeyUIHandler) rekeyNeeded(ctx context.Context, item gregor.Item) (err
 		return r.G().GregorDismisser.DismissItem(item.Metadata().MsgID())
 	}
 
+	if currentDeviceSolvesProblemSet(r.G(), problemSet) {
```
Comment (patrickxb) [inline go/service/rekey_ui_handler.go]: should we dismiss the gregor notification?  or wait until the rekey happens?

Comment (maxtaco) [inline go/service/rekey_ui_handler.go]: I think the latter, we need to wait....

File: go/service/rekey_ui_handler.go [hunk 2]
```diff
@@ -103,9 +103,54 @@ func (r *RekeyUIHandler) rekeyNeeded(ctx context.Context, item gregor.Item) (err
...
+	for _, kid := range tlf.Solution_kids {
+		if kid.Equal(ourKid) {
+			return true
+		}
+	}
+	return false
+}
+
+// currentDeviceSolvesProblemSet returns true if the current device can key all
+// of the folders in the ProblemSet are solved by the current device's key.
```
Comment (maxtaco) [inline go/service/rekey_ui_handler.go]: this comment doesn't parse, i will change it.

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**